Apps/Sandcastle/gallery/Custom Projection.html

```diff
@@ -0,0 +1,101 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no">
+    <meta name="description" content="Demonstration of Custom projections.">
+    <meta name="cesium-sandcastle-labels" content="Showcases">
+    <title>Cesium Demo</title>
+    <script type="text/javascript" src="../Sandcastle-header.js"></script>
+    <script type="text/javascript" src="../../../ThirdParty/requirejs-2.1.20/require.js"></script>
+    <script type="text/javascript">
+        if(typeof require === "function") {
+            require.config({
+                baseUrl : '../../../Source',
+                waitSeconds : 120
+            });
+        }
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+
+var projectionText =
+    'function createProjectionFunctions(callback) {\n' +
+    '     function project(cartographic, result) {\n' +
+    '          result.x = cartographic.longitude * 6378137.0;\n' +
+    '          result.y = cartographic.latitude * 2.0 * 6378137.0;\n' +
+    '          result.z = cartographic.height + Math.abs(cartographic.longitude * cartographic.longitude * 1000000.0);\n' +
+    '     }\n' +
+    '     function unproject(cartesian, result) {\n' +
+    '          result.longitude = cartesian.x / 6378137.0;\n' +
+    '          result.latitude = cartesian.y / (2.0 * 6378137.0);\n' +
+    '          result.height = cartesian.z - Math.abs(result.longitude * result.longitude * 1000000.0);\n' +
+    '     }\n' +
+    '     callback(project, unproject);\n' +
+    ' }\n';
+var projectionTextUrl = 'data:text/plain;base64,' + window.btoa(projectionText);
+
+if (Cesium.FeatureDetection.isInternetExplorer()) {
+    projectionTextUrl = '../../SampleData/SampleProjection.txt';
+}
+
+var customProjection = new Cesium.CustomProjection(projectionTextUrl, 'sample projection');
+customProjection.readyPromise
```

Comment:
I think this logic should be inside scene. Have the scene wait for the projection to be ready before it starts to render. This would also work when using proj4js and have Proj4Projection load the module.

Reply:
Looks like there are several `project` calls before the first Scene update, including but maybe not limited to:
* `Camera` constructor
* `ScreenSpaceCameraController` constructor
* scene morph from setting `sceneMode` when initializating the viewer

I'm not sure if all of that can be modified, or if there's anything else users would like to do that would break if the projection isn't ready.